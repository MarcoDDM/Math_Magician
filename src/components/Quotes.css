.quotes {
  justify-content: center;
  align-items: center;
  margin: auto;
}

.quotes p {
  color: black;
  font-size: 2rem;
}

.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: auto;
  color: rgb(214, 125, 0);
  font-size: 2rem;
}
