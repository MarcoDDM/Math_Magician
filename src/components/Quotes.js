import { useState, useEffect } from 'react';

function Quotes() {
  const [quotes, setQuotes] = useState([]);

  useEffect(() => {
    const category = 'success';
    const fetchData = async () => {
      const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
        headers: {
          'X-Api-Key': '4ClfOE/97HHOf/lDFSpIJA==JqOaqEP6vNaKD1HF',
        },
      });
      const data = await res.json();
      setQuotes(data);
    };
    fetchData();
  }, []);

  return (
    <ul>
      {quotes.map((quote) => (
        <div key={quote.id} className="quotes">
          <p>
            {quote.quote}
            <br />
            -
            {quote.author}
          </p>
        </div>
      ))}
    </ul>
  );
}

export default Quotes;
