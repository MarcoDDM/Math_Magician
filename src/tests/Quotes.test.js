import { render, screen } from '@testing-library/react';
import Quotes from '../components/Quotes';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Quote component', () => {
  let quote;
  beforeEach(() => {
    render(<Quotes />);
    quote = screen.getByTestId('quote');
  });

  it('Should test if Quote component is inside the document', () => {
    expect(quote).toBeInTheDocument();
  });

  it('Should render a snapshot of the Quote component', () => {
    expect(quote).toMatchSnapshot();
  });
});