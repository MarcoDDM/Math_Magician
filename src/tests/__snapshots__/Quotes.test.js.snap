// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Quote component Should render a snapshot of the Quote component 1`] = `
<div
  class="loading"
  data-testid="quote"
>
  Loading...
</div>
`;
